core/node/proof_data_handler/src/tee_request_processor.rs

```diff
@@ -41,32 +38,71 @@ impl TeeRequestProcessor {
     ) -> Result<Json<TeeProofGenerationDataResponse>, RequestProcessorError> {
         tracing::info!("Received request for proof generation data: {:?}", request);
 
-        let mut connection = self
-            .pool
+        let mut min_batch_number: Option<L1BatchNumber> = None;
+
+        loop {
+            let l1_batch_number = match self
+                .lock_batch_for_proving(request.tee_type, min_batch_number)
+                .await?
+            {
+                Some(number) => number,
+                None => return Ok(Json(TeeProofGenerationDataResponse(None))),
+            };
+
+            match self.blob_store.get(l1_batch_number).await {
+                Ok(input) => {
+                    return Ok(Json(TeeProofGenerationDataResponse(Some(Box::new(input)))));
+                }
+                Err(ObjectStoreError::KeyNotFound(_)) => {
+                    tracing::warn!(
+                        "Blob for batch number {} has not been found in the object store.",
+                        l1_batch_number
+                    );
+                    self.unlock_batch(l1_batch_number, request.tee_type).await?;
+                    min_batch_number =
+                        min_batch_number.map_or(Some(l1_batch_number + 1), |num| Some(num + 1));
```

Comment:
Nit: You can simplify the RHS to `Some(min_batch_number.unwrap_or(l1_batch_number) + 1)`.